core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -2482,6 +2482,41 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
     assertThat(inputRef1.getIdentifier(), is(inputRef2.getIdentifier()));
   }
 
+  @Test void testExpressionLineageConjuntiveExpression() {
+    // empno is column 0 in catalog.sales.emp
+    // ename is column 1 in catalog.sales.emp
+    // deptno is column 7 in catalog.sales.emp
+    final RelNode rel = convertSql("select (empno = 1 or ename = 'abc') and deptno > 1 from emp");
+    final RelMetadataQuery mq = rel.getCluster().getMetadataQuery();
+
+    final RexNode ref = RexInputRef.of(0, rel.getRowType().getFieldList());
+    final Set<RexNode> r = mq.getExpressionLineage(rel, ref);
+    assertThat(r.size(), is(1));
```

Comment:
It is sad when an assertion like this fails. It says like "expected 1 got 2", and you have absolutely no clue why 1 was expected, what caused 2 and so on.

`assert(size)` before asserting collection contents is almost always a bad idea. There are corner cases like `assertAll`, soft asserts, etc, however, you are not using them here.

In this case, I would suggest removing `assert(size` and use `set.toString()` with the relevant assertion message.

Reply:
I have replaced the assertion with the use of `Iterables.getOnlyElement`, that is: `String actualExp = Iterables.getOnlyElement(r).toString();`

`Iterables.getOnlyElement` states explicitly that we were expecting a single element and we got a different number, I think it conveys the intention in case of failures and it keeps the test compact.

Reply:
`Iterables.getOnlyElement` would be slightly better, however, I believe it would not convey the reason for "only one element expected" (==input SQL would be missing, etc, etc)

Reply:
I have merged the two assertions since they would be overlapping and provided a more comprehensive error message:

```
Lineage for expr '$0' in node 'rel#2:LogicalProject.(input=LogicalTableScan#1,exprs=[AND(OR(=($0, 1), =($1, 'abc')), >($7, 1))])' for query 'select (empno = 1 or ename = 'abc') and deptno > 1 from emp':
'empno' is column 0 in 'catalog.sales.emp', 'ename' is column 1 in 'catalog.sales.emp', and 'deptno' is column 7 in 'catalog.sales.emp' ==> expected: <AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1))]> but was: <[AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1))]>
Expected :AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1))]
Actual   :[AND(OR(=([CATALOG, SALES, EMP].#0.$0, 1), =([CATALOG, SALES, EMP].#0.$1, 'abc')), >([CATALOG, SALES, EMP].#0.$7, 1))]
```